test/contracts/OVM/bridge/OVM_L1CrossDomainMessenger.spec.ts

```diff
@@ -34,6 +34,20 @@ const getXDomainCalldata = (
   ).encodeFunctionData('relayMessage', [target, sender, message, messageNonce])
 }
 
+const deployProxyXDomainMessenger = async (
+  addressManager: Contract,
+  l1XDomainMessenger: Contract
+): Promise<Contract> => {
+  await addressManager.setAddress(
+    'OVM_L1CrossDomainMessenger',
+    l1XDomainMessenger.address
+  )
+  const proxy = await (await ethers.getContractFactory(
+      'Lib_ResolvedDelegateProxy'
+  )).deploy(addressManager.address, 'OVM_L1CrossDomainMessenger')
+  return l1XDomainMessenger.attach(proxy.address)
```

Comment:
Just double checking that this will just be a contract using the L1CrossDomainMessenger interface, but the address of the proxy?

Reply:
Yep exactly! I didn't do this at first & instead constructed the bytecode to send manually, but then realized that hilariously because the proxy just forwards everything along, I can just highjack the normal Ethers contract objects to make calls to the proxy.